tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +223,59 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+func UploadGcsObject(localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create storage client: %v", err)
+	}
+	defer client.Close()
+
+	// Open the local file for reading
+	f, err := os.Open(localPath)
+	if err != nil {
+		return fmt.Errorf("failed to open local file: %v", err)
+	}
+	defer f.Close()
+
+	if err != nil {
+		return err
+	}
+
+	// Create a writer to upload the object
+	obj := client.Bucket(bucketName).Object(objectName)
+	w := obj.NewWriter(ctx)
+
+	// Set content encoding if gzip compression is needed
+	if uploadGzipEncoded {
+
+		// Create a gzip writer on top of the object writer
+		gw := gzip.NewWriter(w)
```

Comment:
See if you can directly reuse that function here (or if needed, move that utility from there to util/file_operations.go for reuse here) rather than duplicating the logic here, which is always reason for double the maintenance.

Reply:
> I am not sure that gzip.NewWriter would work for an object-writer created from a GCS client

If it does work, then you can leave it as it is. Still see if you can merge the two utilities int o one or at least reduce the logic duplication.